pyvortex/src/dataset.rs

```diff
@@ -78,32 +65,37 @@ fn projection_from_python(columns: Option<Vec<Bound<PyAny>>>) -> PyResult<Projec
     }
 
     Ok(match columns {
-        None => Projection::All,
-        Some(columns) => Projection::Flat(
-            columns
+        None => ident(),
+        Some(columns) => {
+            let fields = columns
                 .iter()
                 .map(field_from_pyany)
-                .collect::<PyResult<Vec<FieldName>>>()?,
-        ),
+                .collect::<PyResult<Arc<[FieldName]>>>()?;
+
+            Select::include_expr(fields, ident())
```

Comment:
@AdamGS I found that anything that's not covered by https://github.com/spiraldb/vortex/blob/cb11b1e3f1cefa684bf68d7985ad8794378eee21/vortex-expr/src/datafusion.rs#L14 will likely yield a broken read

Reply:
However, I have found select to work in rust tests

Reply:
Ignore me, I see that you changed the translator from manual pack to select